src/SystemManager.cc

```diff
@@ -159,26 +244,35 @@ void SystemManager::Reset(const UpdateInfo &_info, EntityComponentManager &_ecm)
           << system.name << "]\n"
           << "Systems created without plugins that do not implement Reset"
           << " will not be reloaded. Reset may not work correctly\n";
+        index++;
```

Comment:
nit (below as well):
```suggestion
        ++index;
```